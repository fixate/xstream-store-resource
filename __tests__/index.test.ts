import createActionTypes from '../src/action-types';
import createResource from '../src/index';

describe('xstream-store-resource', () => {
  test('-> throws if initialised without name', () => {
    expect(() => createResource({name: '', url: 'fakeurl'})).toThrow();
  });

  test('-> throws if initialised without url', () => {
    expect(() => createResource({name: 'fakename', url: ''})).toThrow();
  });

  test('-> configures request, success, and failure actions', () => {
    expect(false).toBe(true);
  });

  test('-> dispatches actions when services respond', () => {
    expect(false).toBe(true);
  });

  test('-> handles endpoints with multiple parameters', () => {
    expect(false).toBe(true);
  });

  test('-> accepts a stream creator that extends the resources stream', () => {
    expect(false).toBe(true);
  });

  test('-> returns actions and action types', () => {
    const {actionTypes, actions} = createResource({name: 'my-service', url: 'fake'});

    expect(actionTypes).toMatchSnapshot();
    expect(actions).toMatchSnapshot();
  });
});
