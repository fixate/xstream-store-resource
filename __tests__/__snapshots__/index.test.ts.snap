// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xstream-store-resource -> returns actions and action types 1`] = `
Object {
  "CREATE": "@my-service/create",
  "CREATE_FAILURE": "@my-service/create-failure",
  "CREATE_SUCCESS": "@my-service/create-success",
  "FIND": "@my-service/find",
  "FIND_FAILURE": "@my-service/find-failure",
  "FIND_SUCCESS": "@my-service/find-success",
  "GET": "@my-service/get",
  "GET_FAILURE": "@my-service/get-failure",
  "GET_SUCCESS": "@my-service/get-success",
  "PATCH": "@my-service/patch",
  "PATCH_FAILURE": "@my-service/patch-failure",
  "PATCH_SUCCESS": "@my-service/patch-success",
  "REMOVE": "@my-service/remove",
  "REMOVE_FAILURE": "@my-service/remove-failure",
  "REMOVE_SUCCESS": "@my-service/remove-success",
  "RESET": "my-service/reset",
  "UPDATE": "@my-service/update",
  "UPDATE_FAILURE": "@my-service/update-failure",
  "UPDATE_SUCCESS": "@my-service/update-success",
}
`;

exports[`xstream-store-resource -> returns actions and action types 2`] = `
Object {
  "create": [Function],
  "createFailure": [Function],
  "createSuccess": [Function],
  "find": [Function],
  "findFailure": [Function],
  "findSuccess": [Function],
  "get": [Function],
  "getFailure": [Function],
  "getSuccess": [Function],
  "patch": [Function],
  "patchFailure": [Function],
  "patchSuccess": [Function],
  "remove": [Function],
  "removeFailure": [Function],
  "removeSuccess": [Function],
  "reset": [Function],
  "update": [Function],
  "updateFailure": [Function],
  "updateSuccess": [Function],
}
`;
